src/index.js

```diff
@@ -148,14 +125,26 @@ function getInterface(proto) {
  */
 function getPropertyNames(proto) {
     const results = []
-
     while (proto !== Object.prototype) {
         Object.getOwnPropertyNames(proto).reduce((arr,k) => arr.indexOf(k) === -1 ? arr.push(k) && arr : arr, results)
         proto = Object.getPrototypeOf(proto).constructor.prototype
     }
-
     return results
 }
 
+function isMixin(x) {
+    return (typeof x == 'function') && !!x.mixin
+}
+
+function isClass(x) {
+    return (typeof x == 'function') && (s => /^class\s/.test(s)
+        || /^.*classCallCheck\(/.test(s.replace(/^[^{]*{\s*/,'').replace(/\s*}[^}]*$/,'')))(x.toString())
```

Comment:
What purpose does that arrow function serve? Why not assign `const s = x.toString()` and execute that stuff directly in the scope of `isClass()`?

Also, imho, creating such a long expression is hard to read. If this is about saving bytes, then leave it to Babel :) Keep the source clean and easy to read and let the transpiler do the dirty work!